examples/pytorch/image-classification/run_image_classification.py

```diff
@@ -291,19 +291,24 @@ def compute_metrics(p):
     )
 
     # Define torchvision transforms to be applied to each image.
+    if "shortest_edge" in feature_extractor.size:
+        size = feature_extractor.size["shortest_edge"]
+    else:
+        size = (feature_extractor.size["height"], feature_extractor.size["width"])
     normalize = Normalize(mean=feature_extractor.image_mean, std=feature_extractor.image_std)
     _train_transforms = Compose(
         [
-            RandomResizedCrop(feature_extractor.size),
+            RandomResizedCrop(size),
+            Resize(size),
```

Comment:
I'm not sure this makes sense, usually one first resizes the shortest edge, then do a center crop.

Here it's the other way around. I would perhaps swap them here?

Reply:
Also, this potentially leads to images that aren't batchable. This is because, when providing an integer size argument to torchvision's Resize, it will only resize the smaller edge.

This means that you are left with images that have different sizes, making batching not possible (unless you pad them).

Reply:
This was a mistake on my part - I didn't mean to add in `Resize` here - I've removed it now. It should now be OK as `RandomResizedCrop` will resize the cropped image to `(size[0], size[1])` if `size` is a tuple and `(size, size)` if size is an int.